contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -47,448 +40,274 @@ contract FraudProofFacet is StateChannelCommon {
+            blockInvalidSTProof.invalidBlock.signature
+        );
+
+        if(fraudProofVerificationContext.channelId != fraudBlock.transaction.header.channelId) {
+            revert ErrorNotSameChannelId();
+        }
         
-    //     if(dispute.channelId != fraudBlock.transaction.header.channelId) {
-    //         slashParticipants[0] = dispute.disputer;
-    //         return (false, slashParticipants, abi.encode("BLOCK INVALID STATE TRANSITION: CHANNEL ID MISMATCH"));
-    //     }
-
-    //     (bool isTransitionValid, bytes memory encodedModifiedState) = executeStateTransitionOnState(
-    //         fraudBlock.transaction.header.channelId,
-    //         blockInvalidSTProof.encodedState,
-    //         fraudBlock.transaction
-    //     );
-
-    //     if (!isTransitionValid) {
-    //         slashParticipants[0] = dispute.disputer;
-    //         return (false, slashParticipants, new bytes(0));
-    //     }        
-    //     if (keccak256(encodedModifiedState) != dispute.latestStateSnapshotHash) {
-    //         slashParticipants[0] = dispute.disputer;
-    //         return (false, slashParticipants, abi.encode("BLOCK INVALID STATE TRANSITION: STATE HASH MISMATCH"));
-    //     }
-    //     address[] memory returnedSlashParticipants = _collectBlockConfirmationAddresses(
-    //         blockInvalidSTProof.fraudBlockConfirmation.signedBlock.encodedBlock,
-    //         blockInvalidSTProof.fraudBlockConfirmation.signatures
-    //     );
-    //     return (true, returnedSlashParticipants, new bytes(0));
-    // }
-
-    // function _verifyBlockDoubleSign( 
-    //     Dispute memory dispute,
-    //     Proof memory proof
-    // ) internal returns (bool isValid, address[] memory slashParticipants, bytes memory fraudProofErrorResult) {
-
-    //     BlockDoubleSignProof memory blockDoubleSignProof = abi.decode(proof.encodedProof, (BlockDoubleSignProof));
-
-    //     Block memory block1 = abi.decode(blockDoubleSignProof.block1.encodedBlock, (Block));
-    //     Block memory block2 = abi.decode(blockDoubleSignProof.block2.encodedBlock, (Block));
-
-    //     address[] memory slashParticipants = new address[](1);
-    //     if(dispute.channelId != block1.transaction.header.channelId || dispute.channelId != block2.transaction.header.channelId) {
-    //         slashParticipants[0] = dispute.disputer;
-    //         return (false, slashParticipants, abi.encode("BLOCK DOUBLE SIGN: CHANNEL ID MISMATCH"));
-    //     }
-
-    //     if(block1.stateSnapshotHash != block2.stateSnapshotHash && block1.previousBlockHash != block2.previousBlockHash) {
-    //         slashParticipants[0] = dispute.disputer;
-    //         return (false, slashParticipants, abi.encode("BLOCK DOUBLE SIGN: STATE HASH MISMATCH"));
-    //     }
+        if(fraudBlock.transaction.header.transactionCnt == 0){
+            require(fraudBlock.previousBlockHash == keccak256(abi.encode(fraudBlock))); 
```

Comment:
for blockHeight 0, the previousBlockHash is the hash of the genesisSnapshot -> hash(previousStateSnapshot)